src/movies.js

```diff
@@ -1,35 +1,69 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+//console.log('movies: ', movies);
 
+const movies = require('./data');
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  return array.map((movie) => movie.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  return array.filter(
+    (movie) =>
+      movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
+  ).length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if (array.length === 0) {
+    return 0;
+  }
+  let mScore = array.reduce(function (acc, value) {
+    if (!value.score) {
+      return acc;
+    }
+    return acc + value.score;
+  }, 0);
+  let mScoreAvg = mScore / array.length;
+  return Math.round(mScoreAvg * 100) / 100;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(array) {
+  let drama = array.filter((movie) => movie.genre.includes('Drama'));
+  return scoresAverage(drama);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(array) {
+  let movieSorted = [...array].sort(function (a, b) {
+    if (a.year < b.year) return -1;
+    if (a.year > b.year) return 1;
```

Comment:
The localeCompare() already returns -1/0/1. For numeric sorting you should use a.something - b.something